domain-management/src/main/java/org/jboss/as/domain/management/security/SecurityRealmService.java

```diff
@@ -570,6 +577,76 @@ public SaslAuthenticationFactory getSaslAuthenticationFactory() {
         return saslAuthenticationFactory;
     }
 
+    @Override
+    public SaslAuthenticationFactory getSaslAuthenticationFactory(final String[] mechanismNames, final Boolean policyNoanonymous) throws StartException {
```

Comment:
This seems like the wrong place for this logic to me.  Why not put it in the Remoting subsystem directly, below?